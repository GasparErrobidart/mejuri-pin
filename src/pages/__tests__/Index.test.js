import '@testing-library/jest-dom'
import 'jest-styled-components'

import configureStore    from 'redux-mock-store'
import React             from 'react'
import { cleanup }       from '@testing-library/react'
import { Provider }      from 'react-redux'
import renderer          from 'react-test-renderer'
import {ThemeProvider}   from 'styled-components'

import Index             from '../Index'
import theme             from '../../components/GlobalStyle/themes/mejuri-pin'

const mockStore = configureStore([]);


describe('Index component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
        filter : [],
        likes : [],
        categories : { data :
          {
            "0" : { name : "test category" , id: "0", products : []}
          }
        },
        products : { data : {} }
    });
  });

  it("Should render",()=>{
    const component       = renderer.create(
      <Provider store={store}>
        <ThemeProvider theme={theme}>
          <Index magicCategoryId="0"/>
        </ThemeProvider>
      </Provider>
    );
    const tree            = component.toJSON()
    expect(tree).toMatchSnapshot();
  })

  afterEach(cleanup)


});
