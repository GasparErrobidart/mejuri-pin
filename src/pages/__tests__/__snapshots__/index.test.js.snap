// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index component Should render 1`] = `
.c0 {
  padding: 0 20px;
  box-sizing: border-box;
  width: 100%;
}

.c3 {
  height: 50px;
  padding: 5px 0;
}

.c3 > img {
  height: 100%;
  width: auto;
}

.c2 {
  font-family: 'Lato',sans-serif;
  font-size: 30px;
  text-align: left;
  position: relative;
  cursor: pointer;
}

.c2 *[animated-heart] {
  color: red;
  position: absolute;
  opacity: 0;
}

.c2 *[animated-heart].animating {
  -webkit-animation: pop 700ms;
  animation: pop 700ms;
}

.c2 > span {
  position: absolute;
  font-size: 12px;
  display: inline-block;
  padding: 2px;
  font-weight: 400;
  margin-left: 22px;
  margin-top: 10px;
  text-align: center;
  border-radius: 2px;
  background: #888888;
  color: white;
  left: 0;
  top: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 > * {
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > *:nth-child(odd) {
  -webkit-flex: 0 0 55px;
  -ms-flex: 0 0 55px;
  flex: 0 0 55px;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c1 > *:nth-child(2) {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.c11 {
  height: 1px;
  opacity: 0.0;
}

.c10 {
  display: grid;
  grid-template-columns: repeat( auto-fit,minmax(20vh,40vh) );
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  grid-gap: 20px;
  margin: 30px 10px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-align-content: start;
  -ms-flex-line-pack: start;
  align-content: start;
}

.c10 > * {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  position: fixed;
  z-index: 1;
  top: 60px;
  left: 0;
  width: 100%;
  -webkit-transition: opacity 300ms ease-in;
  transition: opacity 300ms ease-in;
  opacity: 0.2;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 > *:first-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c8 {
  overflow-y: scroll;
  max-height: 50vh;
  border-top: 1px solid #F3F3F3;
}

.c6 {
  background: white;
  padding: 15px 0;
  border-top: 1px solid #F3F3F3;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 30px;
  cursor: pointer;
}

.c9:hover,
.c9:active {
  border-color: #CCCCCC;
}

.c9 * {
  -webkit-transition: all 200ms ease-in;
  transition: all 200ms ease-in;
  color: #888888;
}

.c9:hover * {
  color: #303030;
}

.c9 div:first-child {
  overflow: hidden;
  -webkit-flex: 0 0 0;
  -ms-flex: 0 0 0;
  flex: 0 0 0;
  height: 15px;
}

@media (min-width:768px) {
  .c10 {
    margin: 40px;
    grid-gap: 40px;
  }
}

@media (min-width:768px) {
  .c4 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }

  .c4 > *:first-child {
    padding-top: 30px;
    -webkit-flex: 0 0 250px;
    -ms-flex: 0 0 250px;
    flex: 0 0 250px;
  }

  .c4 > *:last-child {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media (min-width:768px) {
  .c5 {
    position: static;
    top: auto;
    left: auto;
    width: auto;
  }
}

@media (min-width:768px) {
  .c8 {
    max-height: auto;
    overflow-y: visible;
    border: none;
  }
}

@media (min-width:768px) {
  .c6 {
    background: none;
  }
}

<div
  className="app-wrapper"
>
  <header
    className="main-header"
  >
    <div
      className="c0"
    >
      <div
        className="c1"
      >
        <div
          className="c2"
          onClick={[Function]}
        >
          <div
            animated-heart="true"
            className=""
            onAnimationEnd={[Function]}
          >
            <svg
              fill="none"
              height="1em"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
              />
            </svg>
          </div>
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
            />
          </svg>
          <span>
            0
          </span>
        </div>
        <div>
          <div>
            <div
              className="c3"
            >
              <img
                alt="Mejuri Logo"
                src="/mejuri.svg"
              />
            </div>
          </div>
        </div>
        <div />
      </div>
    </div>
  </header>
  <main
    className="main-content"
  >
    <div
      className="c4"
    >
      <div>
        <div
          className="c5"
          disabled={true}
        >
          <div
            className="c6"
          >
            <div
              className="c0"
            >
              <div
                className="c7"
              >
                <div
                  onClick={[Function]}
                >
                  Filter 
                   
                  <svg
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeWidth="2"
                    style={
                      Object {
                        "color": undefined,
                      }
                    }
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polyline
                      points="18 15 12 9 6 15"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            className="c8"
          >
            <div
              className="c6"
              onClick={[Function]}
            >
              <div
                className="c9"
                onClick={[Function]}
              >
                <div />
                <div>
                  Liked
                </div>
              </div>
            </div>
            <div
              className="c6"
              onClick={[Function]}
            >
              <div
                className="c9"
                onClick={[Function]}
              >
                <div />
                <div>
                  test category
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div
          className="c10"
        />
        <div>
          <div
            className="c11"
          />
        </div>
      </div>
    </div>
  </main>
  <footer
    className="main-footer"
  >
    <div
      className="c0"
    />
  </footer>
</div>
`;
